packages/payment-processor/src/payment/index.ts

```diff
@@ -25,28 +32,47 @@ export class UnsupportedNetworkError extends Error {
   }
 }
 
+/**
+ * Details required for a token swap:
+ *
+ *  - maxInputAmount: maximum number of ERC20 allowed for the swap before payment, considering both amount and fees
+ *  - path: array of token addresses to be used for the "swap path".
+ *    ['0xPaymentCurrency', '0xIntermediate1', ..., '0xRequestCurrency']
+ *  - deadline: time in milliseconds since UNIX epoch, after which the swap should not be executed.
+ */
+export interface ISwapSettings {
+  deadline: number;
+  maxInputAmount: BigNumber;
+  path: string[];
+}
+
 /**
  * Processes a transaction to pay a Request.
  * Supported networks: ERC20_PROXY_CONTRACT, ETH_INPUT_DATA, ERC20_FEE_PROXY_CONTRACT
  *
- * @throws UnsupportedNetworkError if network isn't supported
+ * @throws UnsupportedNetworkError if network isn't supported for swap or payment.
  * @param request the request to pay.
  * @param signerOrProvider the Web3 provider, or signer. Defaults to window.ethereum.
  * @param amount optionally, the amount to pay. Defaults to remaining amount of the request.
+ * @param swapSettings optionally, the information of how to swap from another payment token.
  * @param overrides optionally, override default transaction values, like gas.
  */
 export async function payRequest(
   request: ClientTypes.IRequestData,
   signerOrProvider: Web3Provider | Signer = getProvider(),
   amount?: BigNumberish,
+  swapSettings?: ISwapSettings,
```

Comment:
changing the order of the parameters makes this PR a breaking change.

Reply:
It also makes it pretty dependant to a payment method, which is kind of the opposite of the desired behaviour of this generic `payRequest` method... 

an idea would be to allow `swapSettings` on `overrides`? 
Something like this:
`overrides?: ITransactionOverrides | ISwapSettings`. Or, to be even more generic, `overrides?: ITransactionOverrides  | {swapSettings: ISwapSettings}`

Reply:
I'll go with your suggestion above and make it a different function then.